src/read.jl

```diff
@@ -74,34 +74,38 @@ const tags = Dict{Symbol,Function}()
 
 const raise = Dict{Symbol,Function}()
 
-function _raise_recursive(d::AbstractDict, cache)
+function _raise_recursive(d::AbstractDict, cache, init)
   if haskey(d, :tag) && haskey(tags, Symbol(d[:tag]))
-    cache[d] = tags[Symbol(d[:tag])](applychildren!(x -> raise_recursive(x, cache), d))
+    if Symbol(d[:tag]) in (:ref, :datatype)
```

Comment:
Can we obviate the need for this check and pass init generically here?

Reply:
It would mean every `tag[X]` needs to be a function of two arguments `(d, init)` where `init` is unused for a bunch of cases. Personally, I think the check is cleaner, but I can change it if you like.

Reply:
So it might be polluting, hmm.

The init generally seems to be a way to qualify the symbol anyway. I guess one doesn't usually care beyond a type or global ref.

Reply:
@DhairyaLGandhi so is this okay to keep? If so, I fixed the other comment about the tests.